docs/DeveloperGuide.md

```diff
@@ -1,34 +1,630 @@
+     e.g. `'-e '`
+   * Ending option should be in the form of `' <option>'` with 1 leading whitespace and no trailing whitespace.
+     e.g. `' -e'`
+   * During processing, next option should be in the form of `' <option> '` with 1 leading and trailing whitespace.
+     e.g. `' -e '`
+3. As rearrangement of options is allowed, option detection should cater to non-fixed option order. 
+   Apache Commons Lang, 3.11, providing the StringUtils class is used to cater to consideration 2.
+   * StringUtils.startsWithAny() - detection of start option with non-fixed order.
+   * StringUtils.endsWithAny()   - detection of end option with non-fixed order.
+   * StringUtils.indexOfAny()    - detection of during processing option with non-fixed order.
+4. As multiple whitespaces is allowed, options and arguments should be fully trimmed (leading and trailing).
+5. As the ArrayList<String> is passed back to the main program and is being used by CommandHandler,
+   the argument field should compulsory and appended with empty string if empty to facilitate validations and option-argument
+   pairwise logic.
+6. getParseInput should always return a new ArrayList<String> per new input.
+
+
+### 3.4 CommandHandler Component
+![CommandHandlerClassDiagram](img/CommandHandlerClassDiagram.png)\
+_Figure 6: CommandHandler Class Diagram_\
+![CommandHandlerSequenceDiagram](img/CommandHandlerSequenceDiagram.png)\
+_Figure 7: CommandHandler Sequence Diagram_
+
+#### Description
+The `CommandHandler` component is the object class itself.
+The role of the `CommandHandler` is to convert `parsedArguments` from the `ParserHandler`
+into subsequent `Command` objects which will be executed thereafter.
+The `XYZCommand` in _Figure 7_ represents a valid command word (e.g. `"xyz"`) from the input
+parsed earlier from the `ParserHandler`. More on the actual valid commands, please refer to
+the [Command Component](#35-command-component) below.
+
+#### Design
+It functions as a mapping from `parsedArguments[0]` to a set of predefined command words 
+(the actual commands of the Application).\
+With reference to the above _Figure 7_, 
+In the case of `"xyz"` being mapped to `XYZCommand`, depending on the other data in the 
+`parsedArguments`,
+* arguments are valid: the `XYZCommand` object is successfully returned.
+* arguments are erroneous/invalid: `CommandException` is thrown.
+
+If `parsedArguments[0]` maps to an empty String `""`, a `null` is returned, allowing the 
+`Finux` to continue prompting for user input.
+
+Finally, if `parsedArguments[0]` cannot be mapped to any command, a `CommandException` is 
+thrown to `Finux` to handle.
+
+Not stated explicitly in the diagrams, when the `exit` command is entered, the `CommandHandler`
+sets the `isExit = true`, resulting in `Finux` proceeding to call `end()` to exit the Application.
+
+### 3.5 Command Component
+![CommandClassDiagram](img/CommandClassDiagram.png)
```

Comment:
The image is too long so text appear too small. Would it be possible to shift some of the classes to a new line instead of arranging all of them in one line?

Reply:
PlantUML limitations, decided to split the diagram into 2 images.
Thanks!